crates/sui-types/src/signature.rs

```diff
@@ -101,6 +110,90 @@ impl GenericSignature {
     pub fn is_upgraded_multisig(&self) -> bool {
         matches!(self, GenericSignature::MultiSig(_))
     }
+
+    /// Parse [enum CompressedSignature] from trait SuiSignature `flag || sig || pk`.
+    /// This is useful for the MultiSig to combine partial signature into a MultiSig public key.
+    pub fn to_compressed(&self) -> Result<CompressedSignature, SuiError> {
```

Comment:
these two methods are moved out of "enum Signature" that does not account for zkLogin sig